src/orders/components/OrderSummary/OrderSummary.tsx

```diff
@@ -8,26 +9,35 @@ import { OrderValue } from "./OrderValue";
 import { PaymentsSummary } from "./PaymentsSummary";
 import { TransactionsApiButtons } from "./TransactionsApiButtons";
 
-type OrderSummaryWithLegacyApi = {
-  useLegacyPaymentsApi: true;
-  onLegacyPaymentsApiCapture: () => any;
-  onLegacyPaymentsApiRefund: () => any;
-  onLegacyPaymentsApiVoid: () => any;
-};
+type EditableOrderSummary = {
+  isEditable: true;
+  onShippingMethodEdit: () => void;
+  errors?: OrderErrorFragment[];
+} & OrderDiscountContextConsumerProps;
 
-type OrderSummaryWithoutLegacyApi = {
-  useLegacyPaymentsApi?: false;
+type ReadOnlyOrderSummary = {
+  isEditable?: false;
 };
 
 type Props = PropsWithBox<
   {
     order: OrderDetailsFragment;
-    onMarkAsPaid: () => any;
-  } & (OrderSummaryWithLegacyApi | OrderSummaryWithoutLegacyApi)
+    onMarkAsPaid?: () => void;
+    useLegacyPaymentsApi?: boolean;
+    onLegacyPaymentsApiCapture?: () => any;
+    onLegacyPaymentsApiRefund?: () => any;
+    onLegacyPaymentsApiVoid?: () => any;
```

Comment:
dont use "any", if you don't consume response, use `() => void` or `() => unknown`